pkg/shared/utils/utils.go

```diff
@@ -53,6 +58,39 @@ func RunCommand(cmd *exec.Cmd) ([]byte, error) {
 	return outb.Bytes(), nil
 }
 
+func RunCommandAndParseOutputLineByLine(cmd *exec.Cmd, pfn parserFn, resChan chan any, logger *logrus.Entry) error {
+	// Get a pipe to read from standard out
+	r, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get command pipe: %w", err)
+	}
+
+	scanner := bufio.NewScanner(r)
+	// Use the scanner to scan the output line by line and parse it
+	done := make(chan struct{})
+	go func() {
+		// Read line by line and process it
+		for scanner.Scan() {
+			line := scanner.Text()
+			resChan <- pfn(line, logger)
+		}
+		done <- struct{}{}
+	}()
+
+	// Start the command and check for errors
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("failed to start command: %w", err)
+	}
+
+	<-done
```

Comment:
nit: maybe a comment here like `// Wait for all stdout lines to be read`

Reply:
Been thinking and I'm not sure we need this to be a go routine unless we also want to capture any stderr messages in parallel with stdout. As the code is today the scanner could be inlined here.